src/components/CmsContent.ts

```diff
@@ -2,29 +2,42 @@ import isEqual from 'lodash/isEqual';
 import { CreateElement, Component as ComponentType, VNode } from 'vue';
 import { Vue, Component, Prop, Watch } from 'vue-property-decorator';
 
+import { pluginOptions } from '../plugins/cms';
+
 @Component({ name: 'cms-content' })
 export default class CmsContent extends Vue {
   @Prop(String) public html!: string;
   @Prop(Object) public extra!: {};
-  @Prop(Object) public renderContext!: {};
   @Prop(String) public zoneId!: string;
   @Prop({ default: '' }) public tag!: string;
 
   public context: object = {};
 
+  public setupComplete: boolean = false;
+
   private created(): void {
-    this.context = {
-      ...this.extra,
-      ...this.renderContext,
-     };
+    this.context = { ...this.extra };
+  }
+
+  private trackEvent(event: string, props: object): void {
+    pluginOptions.trackAnalytics(event, props);
+  }
+
+  private setup(props: object): void {
+    if (!this.setupComplete) {
+      Object.keys(props).forEach((k): void => {
+        this.$set(this.context, k, props[k]);
+      });
+      this.setupComplete = true;
+    }
   }
 
-  @Watch('renderContext', { deep: true, immediate: true })
+  @Watch('extra', { deep: true, immediate: true })
   private updateContext(value?: object, oldValue?: object): void {
     // Prevent deep watcher for extra from firing too often.
     // See: https://github.com/vuejs/vue/issues/5776
     if (oldValue !== undefined && !isEqual(value, oldValue)) {
-      Object.assign(this.context, this.renderContext);
+      Object.assign(this.context, this.extra);
```

Comment:
Shouldn't we do something like `this.context = { ...this.conext, ...this.extra }` so Vue's reactivity to work?

Reply:
Actually, don't you need to use Vue's special `set` method for this to work?

https://vuejs.org/v2/guide/reactivity.html#For-Objects

Reply:
Right, The current code breaks if a new property is created in the renderContext that didn't previous exist (but should work for changes to existing properties). Mohan's way should work since the entire object would be re-created and made reactive.